src/solver.jl

```diff
@@ -82,7 +82,13 @@ function init_starting_point!(solver::MadNLP.AbstractMadNLPSolver)
     zl .+= 1.0 + delta_s
     zu .+= 1.0 + delta_s
 
-    μ = dot(xl .- lb, zl) + dot(ub .- xu, zu)
+    μ = 0.0
+    if length(zl) > 0
+        μ += dot(xl .- lb, zl)
```

Comment:
We allocate a new vector for `xl .- lb`, we should probably do `dot(xl, zl) - dot(lb, zl)`.